packages/sns/src/types/actions.ts

```diff
@@ -0,0 +1,97 @@
+import type { Principal } from "@dfinity/principal";
+import type {
+  DefaultFollowees,
+  DeregisterDappCanisters,
+  ExecuteGenericNervousSystemFunction,
+  Motion,
+  NeuronPermissionList,
+  RegisterDappCanisters,
+  Subaccount,
+} from "../../candid/sns_governance";
+import type { Option } from "./common";
+
+export type ActionOptional =
```

Comment:
I struggled a lot with the naming.

The new type is called `Option`. Therefore, I called the new types that use it "XxxxOptional".

I'm all ears for better naming.

Reply:
Aaaaah, it means "Action but with everything inside it converted from `[]|[T]` optionals to `undefined | T` optionals.
I read it as `Action | undefined`.

I think this is the normal type and it should have a normal name.
It could even be `Action` (in which case you'd have to import the Action from ../../candid/sns_governance with a different name).
Or `SnsAction` or something like that.
I don't think the name should reference the fact that it has a sane structure because (hopefully) users of the type shouldn't be concerned with that and they just have a better experience.
Or am I missing something about how this will be used?